### PR TITLE
CLDR-13124 avoid replacing xmlsource in makeVettedSource

### PR DESCRIPTION
**Pull request by btangmu**
##### Checklist

- [x] Issue filed: https://unicode-org.atlassian.net/browse/CLDR-13124
- [x] Updated PR title and link in previous line to include Issue number

### CODE REVIEW

**Comment by btangmu** [comment]
I've run into a lot of difficulty trying to test this including the step of generating vxml. I'm making the PR anyway, for the sake of at least enabling some discussion.

Steven, did you save a copy of the last version of srl-vvprobe.jsp you made? I can't find it now.

**Comment by btangmu** [comment]
This morning I was able to test successfully. The bug reproduces on localhost with master branch (just revising outputAllFiles to output only aa.xml to save time and confusion), and doesn't reproduce on localhost with branch t13124_xmlsource.

**Comment by macchiati** [comment]
Sounds great.

{phone}

File: tools/cldr-apps/src/org/unicode/cldr/web/STFactory.java [hunk 1]
```diff
@@ -691,15 +691,26 @@ private boolean isValidSurveyToolVote(UserRegistry.User user, String xpath) {
...
          * Called by PerLocaleData.makeSource(resolve = false), and by PerLocaleData.makeVettedSource.
+         *
+         * TODO: revisit whether resolveMorePaths is still necessary; when added for vxml generation it involved
+         * cases where last-release value made a difference to vote resolution; now that "baseline" = trunk not
+         * last-release it's possible that vote resolution isn't needed for items without current votes.
          */
-        private void loadVoteValues(boolean resolveMorePaths) {
+        private void loadVoteValues(DataBackedSource targetXmlSource, boolean resolveMorePaths, boolean doStampAndListen) {
+            /*
+             * TODO: move the readonly check to the caller
```
Comment (srl295) [inline tools/cldr-apps/src/org/unicode/cldr/web/STFactory.java]: the caller already has a readonly check.

File: tools/cldr-apps/src/org/unicode/cldr/web/STFactory.java [hunk 2]
```diff
@@ -1119,12 +1136,12 @@ public synchronized XMLSource makeSource(boolean resolved) {
...
          * This function is similar to makeSource, but with resolveMorePaths true.
          *
-         * @return this.xmlsource, the new XMLSource for this PerLocaleData.
+         * @return the DataBackedSource (NOT the same as PerLocaleData.xmlsource)
          */
         public synchronized XMLSource makeVettedSource() {
-            xmlsource = new DataBackedSource(this);
-            loadVoteValues(true /* resolveMorePaths */);
-            return xmlsource;
+            DataBackedSource vxmlSource = new DataBackedSource(this);
```
Comment (srl295) [inline tools/cldr-apps/src/org/unicode/cldr/web/STFactory.java]: This is the most important line :)

Comment (srl295) [reply tools/cldr-apps/src/org/unicode/cldr/web/STFactory.java]: :+2: for the new name (vxmlsource)

**Pull request merged**